### ISSUE TITLE
The Karmada cache was not cleared when executed `hack/undeploy-karmada.sh`

### ISSUE THREAD

**Issue by lonelyCZ** [karmada-io/karmada#1042]
**What happened**:
When I redeploy karmada in my cluster, I firstly undeployed karmada by `hack/undeploy-karmada.sh`, then deployed karmada by `hack/remote-up-karmada.sh`, but the previous resources have been redeployed.

**What you expected to happen**:
I want an initialized environment when I redeploy karmada in my host cluster.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
I think that `/var/lib/karmada-etcd` hasn't be deleted when executing `hack/undeploy-karmada-sh`.

Is this on purpose or forgotten to clear？

**Environment**:
- Karmada version: master
- Others:

**Comment by RainbowMango** [comment]
> I think that /var/lib/karmada-etcd hasn't be deleted when executing hack/undeploy-karmada-sh.

I don't even know what's the `karmada-etcd` thing.

**Comment by lonelyCZ** [comment]
It is hostPath of karmada-etcd.
https://github.com/karmada-io/karmada/blob/ca1b1de475bff8bbf4f2a8b221d8f22db9929e91/artifacts/deploy/karmada-etcd.yaml#L80-L82

**Comment by feeltimeQ** [comment]
Hi, I'm not quite sure if the previous resources are karmada's own resources, or resources that were schedue to other clusters by karmada? @lonelyCZ

**Comment by lonelyCZ** [comment]
> I'm not quite sure if the previous resources are karmada's own resources, or resources that were schedue to other clusters by karmada? 

There are resources created by karmada, if we are undeploying karmada that should be deleted if the resource need not to be saved.

**Comment by duanmengkk** [comment]
I have the same problem，should we prompt the user in the shell script, let them manually clear ` /var/lib/karmada-etcd` ? 
Or use some other method to help them clear the dir, this might be a little difficult because IP address of the machine where the script resides may be cannot access to  the IP address of etcd pod located in.

**Comment by duanmengkk** [comment]
Or use sc instead of HostPath？

**Comment by RainbowMango** [comment]
@lonelyCZ How do you say? My general opinion is we should clean the garbage resource thoroughly, and prompt the user on the ones that we can't clear as @duanmengkk mentioned above.

**Comment by lonelyCZ** [comment]
> Or use some other method to help them clear the dir, this might be a little difficult because IP address of the machine where the script resides may be cannot access to the IP address of etcd pod located in. 

Yes, if the `etcd-pod` isn't in the host that `karmada control plane` is deployed, we can't directly delete the directory.

> should we prompt the user in the shell script, let them manually clear /var/lib/karmada-etcd ?

Yes, we can do this.

> Or use StorageClass instead of HostPath?

Is it complicated to introduce StorageClass? After all, script deployment is for quick experience and development, and other highly available deployment ways should be implemented later.

**Comment by duanmengkk** [comment]
> > Or use some other method to help them clear the dir, this might be a little difficult because IP address of the machine where the script resides may be cannot access to the IP address of etcd pod located in.
> 
> Yes, if the `etcd-pod` isn't in the host that `karmada control plane` is deployed, we can't directly delete the directory.
> 
> > should we prompt the user in the shell script, let them manually clear /var/lib/karmada-etcd ?
> 
> Yes, we can do this.
> 
> > Or use StorageClass instead of HostPath?
> 
> Is it complicated to introduce StorageClass? After all, script deployment is for quick experience and development, and other highly available deployment ways should be implemented later.

ok, i will add some prompting in shell

**Comment by lonelyCZ** [comment]
```
[root@master67 karmada]# hack/undeploy-karmada.sh ~/.kube/config karmada-host
Switched to context "karmada-host".
namespace "karmada-system" deleted
deleted cluster karmada-apiserver from /root/.kube/config
Property "users.karmada-apiserver" unset.
deleted context karmada-apiserver from /root/.kube/config
Karmada is undeployed successfully.
Please remove the directory '/var/lib/karmada-etcd' in 10.10.103.67 manually to clean the garbage resource thoroughly.
```
Suggest a blank line

```
Switched to context "karmada-host".
namespace "karmada-system" deleted
deleted cluster karmada-apiserver from /root/.kube/config
Property "users.karmada-apiserver" unset.
deleted context karmada-apiserver from /root/.kube/config

Karmada is undeployed successfully.
Please remove the directory '/var/lib/karmada-etcd' in 10.10.103.67 manually to clean the garbage resource thoroughly.
```
